src/main/java/org/klodnicki/service/BookService.java

```diff
@@ -39,4 +42,25 @@ public void update(BookInfo bookInfo) {
     public List<BookInfo> findBooksByTitleAndAuthor(String title, String author) {
         return bookRepository.findBooksByTitleAndAuthor(title, author);
     }
+
+    public List<String> prepareListOfBorrowedBooksByAccount (Account account) {
+        return prepareListOfBooks(account);
+    }
+
+    private List<String> prepareListOfBooks(Account account) {
+        List<BookInfo> foundBooks = bookRepository.findBooksByAccount(account);
+        List<String> results = new ArrayList<>();
+        for(int i = 0; i< foundBooks.size(); i++) {
+            results.add(foundBooks.get(i).toString());
+        }
+        return results;
+    }
+
+    public List<String> prepareListOfBooks(String title, String author) {
+        List<String> results = new ArrayList<>();
+        for (int i = 0; i < findBooksByTitleAndAuthor(title, author).size(); i++) {
```

Comment:
This is incorrect due to the doubling the select query in database. It is possible for this query to return different lists - that's the first issue - and it also results in weak performance.

Reply:
understood! Changed and ready to be re-viewed!